pkg/ddevapp/hostname_mgt.go

```diff
@@ -98,11 +98,19 @@ func (app *DdevApp) AddHostsEntriesIfNeeded() error {
 
 			// If we had successful lookup and the IP address looked up is local
 			// then we don't have to add it to the /etc/hosts.
-			if err == nil && len(hostIPs) > 0 && netutil.IsLocalIP(hostIPs[0]) {
```

Comment:
why does the original code not already cover it? It looks like `hostIPs` is already an array and even when ipv6 adds another array item: at `hostIPs[0]` can only be a 50/50 chance: either it is `127.0.0.1` or it is `::1`. Is netutil.IsLocalIP not capable of recognizing `::1` as local IP?

Reply:
`netutil.IsLocalIP()` compares the IP with `dockerutil.GetDockerIP()`, which is `127.0.0.1` in this case, and if it's not equal, it compares it with `netutil.GetLocalIPs()` which gets local IPs, but excludes loopback addresses such as `127.0.0.1` and `::1`:

https://github.com/ddev/ddev/blob/cef6f2f7820283abf39e51c8475c2d51b9a33559/pkg/netutil/netutil.go#L47-L96

Theoretically it should work when the IP is resolvable either with IPv4 or IPv6.

I tried to remove this check for loopback `if v.IP.IsLoopback() {`, and tested it with a domain that can only be resolved with IPv6 (with `dnsmasq` help), `ddev start` didn't ask for sudo, but the site didn't open in the browser. I believe that it does not work for us.